pmd-java/src/main/resources/category/java/performance.xml

```diff
@@ -57,6 +57,58 @@ sb.append('a');     // use this instead
         </example>
     </rule>
 
+    <rule name="AvoidApacheCommonsFileItemNonStreaming"
```

Comment:
This looks like any "X method should be avoided on Y type" rule, and doesn't apply to JDK classes. Seems to me maybe a bit too specific of a use-case... I think, at least, that this will become redundant/too specific come 7.0. Improvements to the symbol table and type res will make this trivial to implement in one's own ruleset. So I'm not sure about adding this rule to PMD.

@jsotuyod @adangel  wdyt ?

Reply:
Well… we have had lots of vendor-specific rules over time. We did end up merging / refactoring them together (ie: Log4J / Jakarta / Apache Commons), but not until recently… and we still have JUnit specific rules… JUnit may be extremely popular, but certainly not the only alternative (ie: TestNG)… we also have built-in support for Lombok annotations, and some rules understand Mockito, AssertJ and similar tools.

Even when we refactor and generalize those rules by moving that knowledge to properties, we still keep them in our defaults…

Hence, I see no problem in shipping rules such as this one, and we may refactor and generalize it as we move forward and needs arises.

Moreover, if these rules start to pile up, we may want to start shipping vendor-specific rulesets (ie: `rulesets/vendor/spring.xml`). The fact that we make these rules easier to write is a feature, and something we should push for, but shouldn't necessarily mean that it's up to the user to write them, specially not for the most well known / widely used libraries out there.

Reply:
@jsotuyod Ok then this rule is acceptable. Maybe, we should document this somewhere on the website, for future reference